pkg/kvstore/allocator/allocator.go

```diff
@@ -477,22 +484,14 @@ func (a *Allocator) lockedAllocate(key AllocatorKey) (ID, bool, error) {
 		return 0, false, fmt.Errorf("another writer has allocated this key")
 	}
 
-	lock, err := a.lockPath(k)
-	if err != nil {
-		releaseKeyAndID()
```

Comment:
Shouldn't we still do this?

Reply:
It is moved to the beginning of the function.